client/my-sites/checkout/composite-checkout/hooks/use-effect-on-change.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from 'react';
+
+/*
+ * Run a callback function any time one of the dependencies changes
+ *
+ * This is similar to useEffect, but there are several differences:
+ *
+ * - This hook will not run its callback the first time it is called; it will
+ * only run when one of its dependencies has changed from its previous value.
+ *
+ * - This hook has no cleanup function. The return value of the callback has no
+ * effect.
```

Comment:
Would it be correct to say "The return value of the callback is not used."? For a moment I wondered if normal useEffect callbacks trigger a rerender or some other side effect.